content/en/blog/_posts/2024-09-25-swap-ga-graduation.md

```diff
@@ -0,0 +1,369 @@
+
+---
+layout: blog
+title: "Kubernetes 1.32: GA support for using swap on Linux"
+date: 2024-09-25T10:00:00-08:00
+slug: swap-linux-ga
+---
+
+**Author:** Itamar Holder (Red Hat)
+
+Swap is a fundamental and an invaluable Linux feature.
+It offers numerous benefits, such as effectively increasing a node’s memory by
+swapping out unused data,
+shielding nodes from system-level memory spikes,
+preventing Pods from crashing when they hit their memory limits,
+and [much more](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2400-node-swap/README.md#user-stories).
+As a result, the node special interest group within the Kubernetes project
+has invested significant effort into supporting swap on Linux nodes as a GA feature.
+
+The 1.22 release [introduced Alpha support](/blog/2021/08/09/run-nodes-with-swap-alpha/)
+for configuring swap memory usage for Kubernetes workloads running on Linux on a per-node basis.
+Later, in release 1.28, support for swap on Linux nodes has graduated to Beta, along with many
+new improvements.
+Now, in the 1.32 release, swap is graduating to be Generally Available.
+
+Prior to version 1.22, Kubernetes did not provide support for swap memory on Linux systems.
+This was due to the inherent difficulty in guaranteeing and accounting for pod memory utilization
+when swap memory was involved. As a result, swap support was deemed out of scope in the initial
+design of Kubernetes, and the default behavior of a kubelet was to fail to start if swap memory
+was detected on a node.
+
+In version 1.22, the swap feature for Linux was initially introduced in its Alpha stage.
+This provided Linux users the opportunity to experiment with the swap feature for the first time.
+However, as an Alpha version, it was not fully developed and only partially worked on limited environments.
+
+In version 1.28 swap support on Linux nodes was promoted to Beta.
+The Beta version was a drastic leap forward.
+Not only did it fix a large amount of bugs and made swap work in a stable way,
+but it also brought cgroup v2 support, introduced a wide variety of tests
+which include complex scenarios such as node-level pressure, and more.
+It also brought many exciting new capabilities such as the `LimitedSwap` behavior
+which sets an auto-calculated swap limit to containers, metrics support for tools
+like Prometheus (through the `/metrics/resource` endpoint) and Summary API for
```

Comment:
Rather than list a specific (albeit popular) implementation, are we able to write:
```suggestion
which sets an auto-calculated swap limit to containers, OpenMetrics intrumentation
support (through the `/metrics/resource` endpoint) and Summary API for
```
?